admiral/envs/components/agent.py

```diff
@@ -39,8 +40,14 @@ def __init__(self, action_space=None, **kwargs):
         self.action_space = {} if action_space is None else action_space
 
     def finalize(self, **kwargs):
+        """
+        Wrap all the action spaces with a Dict and seed it if the agent was
+        created with a seed.
+        """
         super().finalize(**kwargs)
         self.action_space = Dict(self.action_space)
+        if self.seed is not None:
```

Comment:
I don't think the 'if ...' is necessary. It makes sense to seed every time and the underlying function can handle 'None' values.